python/lsst/rapid/analysis/butlerUtils.py

```diff
@@ -0,0 +1,688 @@
+        The butler.
+    """
+    # TODO: if/when RFC-811 passes, update this to use the env var
+    # TODO: Add logging to which collections are going in
+    if location not in _LOCATIONS:
+        raise RuntimeError(f'Default butler location only supported for {_LOCATIONS}, got {location}')
+    repodir = LATISS_REPO_LOCATION_MAP[location]
+    LSC = LATISS_SUPPLEMENTAL_COLLECTIONS  # grrr, line lengths
+    collections = (LSC[location] if location in LSC.keys() else []) + LATISS_DEFAULT_COLLECTIONS
+    if extraCollections:
+        collections.extend(extraCollections)
+    return dafButler.Butler(repodir, collections=collections, writeable=writeable, instrument='LATISS')
+
+
+def _repoDirToLocation(repoPath):
+    assert(repoPath in _REPO_PATHS)
+    # can only be in one place, will need changing if we ever have repo
+    # paths that are the same in the repo map
+    repoLocationInverse = {v: k for (k, v) in LATISS_REPO_LOCATION_MAP.items()}
+    assert len(LATISS_REPO_LOCATION_MAP) == len(repoLocationInverse)  # make sure we didn't drop dupes
+    location = repoLocationInverse[repoPath]
+    return location
+
+
+def datasetExists(butler, dataProduct, dataId, **kwargs):
+    """Collapse the tri-state behaviour of butler.datasetExists to a boolean.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        The butler
+
+    dataProduct : `str`
+        The type of data product to check for
+
+    dataId : `dict`
+        The dataId of the dataProduct to check for
+
+    Returns
+    -------
+    exists : `bool`
+        True if the dataProduct exists for the dataProduct and can be retreived
+        else False.
+    """
+    try:
+        exists = butler.datasetExists(dataProduct, dataId, **kwargs)
+        return exists
+    except (LookupError, RuntimeError):
+        return False
+
+
+def sanitize_day_obs(day_obs):
+    """Take string or int day_obs and turn it into the int version.
+
+    Parameters
+    ----------
+    day_obs : `str` or `int`
+        The day_obs to sanitize.
+
+    Returns
+    -------
+    day_obs : `int`
+        The sanitized day_obs.
+
+    Raises
+    ------
+    ValueError
+        Raised if the day_obs fails to translate for any reason."""
+    if isinstance(day_obs, int):
+        return day_obs
+    elif isinstance(day_obs, str):
+        try:
+            return int(day_obs.replace('-', ''))
+        except Exception:
+            ValueError(f'Failed to sanitize {day_obs} to a day_obs')
+    else:
+        raise ValueError(f'Cannot sanitize {day_obs} to a day_obs')
+
+
+def getMostRecentDayObs(butler):
+    """Get the most recent day_obs for which there is data.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler
+        The butler to query.
+
+    Returns
+    -------
+    day_obs : `int`
+        The day_obs.
+    """
+    where = f"exposure.day_obs>{RECENT_DAY}"
+    records = butler.registry.queryDimensionRecords('exposure', where=where, datasets='raw')
```

Comment:
It is now possible to order the results by date (with `.order_by` and also `limit`) so you can now call `next(records)` and immediately get the most recent observation.